_clients/rust.md

```diff
@@ -0,0 +1,379 @@
+---
+layout: default
+title: Rust client
+nav_order: 100
+---
+
+# Rust client
+
+The OpenSearch Rust client lets you connect your Rust application with the data in your OpenSearch cluster. For the client's complete API documentation and more examples, see the [OpenSearch docs.rs documentation](https://docs.rs/opensearch/).
+
+This getting started guide illustrates how to connect to OpenSearch, index documents, and run queries. For the client source code, see the [opensearch-rs repo](https://github.com/opensearch-project/opensearch-rs).
+
+## Setup
+
+If you're starting a new project, add the `opensearch` crate to Cargo.toml:
+
+```rust
+[dependencies]
+opensearch = "1.0.0"
+```
+{% include copy.html %}
+
+Additionally, you may want to add the following `serde` dependencies that help serialize types to JSON and deserialize JSON responses:
+
+```rust
+serde = "~1"
+serde_json = "~1"
+```
+{% include copy.html %}
+
+The Rust client uses the [`reqwest`](https://crates.io/crates/reqwest) higher level HTTP client library for HTTP requests, and reqwest uses the [`tokio`](https://crates.io/crates/tokio) platform to support asynchronous requests. If you are planning to use asynchronous functions, you need to add the `tokio` dependency to Cargo.toml:
+
+```rust
+tokio = { version = "*", features = ["full"] }
+```
+{% include copy.html %}
+
+See the [Sample program](#sample-program) section for the complete Cargo.toml file.
+
+To use the Rust client API, import the modules, structs, and enums you need:
+
+```rust
+use opensearch::OpenSearch;
+```
+{% include copy.html %}
+
+## Connecting to OpenSearch
+
+To connect to the default OpenSearch host, create a default client object that connects to OpenSearch at the address `http://localhost:9200`:
+
+```rust
+let client = OpenSearch::default();
+```
+{% include copy.html %}
+
+To connect to OpenSearch that is running at a different address, create a client with the specified address:
+
+```rust
+let transport = Transport::single_node("http://localhost:9200")?;
+let client = OpenSearch::new(transport);
+```
+{% include copy.html %}
+
+Alternatively, you can customize the URL and use a connection pool by creating a TransportBuilder struct and passing it to OpenSearch::new to create a new instance of the client: 
```

Comment:
```suggestion
Alternatively, you can customize the URL and use a connection pool by creating a TransportBuilder struct and passing it to `OpenSearch::new` to create a new instance of the client: 
```

Reply:
Changed both struct and constructor to have code format.